api/src/main/java/com/epam/pipeline/manager/git/GitReaderApiBuilder.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.manager.git;
+
+import com.epam.pipeline.config.JsonMapper;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import lombok.RequiredArgsConstructor;
+import okhttp3.OkHttpClient;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.util.Assert;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.security.GeneralSecurityException;
+import java.text.SimpleDateFormat;
+import java.util.concurrent.TimeUnit;
+
+@RequiredArgsConstructor
+public class GitReaderApiBuilder {
+
+    private static final String DATA_FORMAT = "yyyy-MM-dd HH:mm:ss Z";
+    private static final int TIMEOUT = 30;
+
+    private final int connectTimeout;
+    private final int readTimeout;
+    private final String apiRoot;
+
+    public GitReaderApiBuilder(final String apiRoot) {
+        this.connectTimeout = TIMEOUT;
+        this.readTimeout = TIMEOUT;
+        this.apiRoot = apiRoot;
+    }
+
+    public GitReaderApi build() {
+        return new Retrofit.Builder()
```

Comment:
It seems that most of this code is duplicate of `GitLabApiBuilder` can we extract common code?